### DISCUSSION TITLE
调整脚注的数字和文本之间的空隙

### DISCUSSION THREAD

**Discussion by earthmoon** [whutug/whu-thesis#111] [Q&A]
你好，
我发现脚注的数字和文字挨得太近了，所以想调整间距，我采用的代码是：

> \usepackage[hang]{footmisc}
\setlength{\footnotemargin}{2mm}

放置在`\begin{document}`之前。
但是运行时报错，为什么呢？

**Comment by tanukihee** [comment]
因为模板中已经配置了相应代码，会有选项冲突，去 cls 里搜自己改吧

**Reply by earthmoon** [reply]
cls代码太专业了

**Comment by sikouhjw** [comment]
模板调用了
https://github.com/whutug/whu-thesis/blob/a0cb5918638fece6c7fa171bb6be179dcf6a3a33/whu-thesis.cls#L426-L427
并且给定了选项 `perpage`，如果你在后面加载 `\usepackage[hang]{footmisc}`，会出现选项冲突，所以需要在
https://github.com/whutug/whu-thesis/blob/a0cb5918638fece6c7fa171bb6be179dcf6a3a33/whu-thesis-demo.tex#L5
之前添加 `\PassOptionsToPackage{hang}{footmisc}`。

**Reply by earthmoon** [reply]
那是不是修改cls更为直接呀？我想在脚注数字和脚注内容之间（页面底部的）添加一个“~”这样的空格，应该怎么修改呢？
下面是关于脚注的代码（我猜的）

> \tl_set:Nn \@makefnmark
>   {
>     \kern \c_zero_dim \textsuperscript { \circled { \@thefnmark } }
>   }
> 
> \RenewDocumentCommand { \@makefntext } { m }
>   {
>     \noindent \hangindent 1 em \circled { \@thefnmark } #1 \hangafter 1
>   }
> 
> \skip_set_eq:NN \headheight \baselineskip
> \skip_set_eq:NN \footskip   \baselineskip
> 
> \dim_set:Nn \footnotesep { 6 pt }
> \setlength { \skip \footins } { \skip_eval:n { 2 \baselineskip } plus 1 fill }
> \tl_set:Nn \footnotesize { \songti \zihao { 5 } }
> \tl_set:Nn \footnoterule
>   { \noindent \rule [ 1 pt ] { 0.3 \columnwidth } { 1 pt } }
> 
> \NewDocumentCommand { \circled } { m }
>   {
>     \resizebox { 1 em } { ! }
>       {
>         \tikz [ baseline = ( char.base ) ]
>           {
>             \node [ shape = circle , draw , inner ~ sep = \c_zero_dim ,
>               minimum ~ size = 1 em ] ( char ) {#1};
>           } % 圆圈数字①
>       }
>   }
